samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -0,0 +1,242 @@
+
+    // Locking to ensure that these aggregated metrics will be created only once across multiple system consumers.
+    synchronized (AGGREGATE) {
+      if (aggEventReadRate == null) {
+        aggEventReadRate = registry.newCounter(AGGREGATE, EVENT_READ_RATE);
+        aggEventByteReadRate = registry.newCounter(AGGREGATE, EVENT_BYTE_READ_RATE);
+        aggReadLatency = new SamzaHistogram(registry, AGGREGATE, READ_LATENCY);
+        aggReadErrors = registry.newCounter(AGGREGATE, READ_ERRORS);
+      }
+    }
+  }
+
+  @Override
+  public void register(SystemStreamPartition systemStreamPartition, String offset) {
+    super.register(systemStreamPartition, offset);
+    String stream = systemStreamPartition.getStream();
+    EventHubEntityConnection connection = connections.get(stream);
+    if (connection == null) {
+      throw new SamzaException("No EventHub connection for " + stream);
+    }
+
+    if (StringUtil.isNullOrWhiteSpace(offset)) {
+      switch (config.getStartPosition(systemStreamPartition.getStream())) {
+        case EARLIEST:
+          offset = START_OF_STREAM;
+          break;
+        case LATEST:
+          offset = END_OF_STREAM;
+          break;
+        default:
+          throw new SamzaException(
+                  "Unknown starting position config " + config.getStartPosition(systemStreamPartition.getStream()));
+      }
+    }
+    connection.addPartition(systemStreamPartition.getPartition().getPartitionId(), offset,
+            new PartitionReceiverHandlerImpl(systemStreamPartition, eventReadRates.get(stream),
+                    eventByteReadRates.get(stream), readLatencies.get(stream), readErrors.get(stream),
+                    serdes.get(stream)));
+  }
+
+  @Override
+  public void start() {
+    connections.values().forEach(EventHubEntityConnection::connectAndStart);
+  }
+
+  @Override
+  public void stop() {
+    connections.values().forEach(EventHubEntityConnection::stop);
+  }
+
+  private class PartitionReceiverHandlerImpl extends PartitionReceiveHandler {
+
+    private final Counter eventReadRate;
+    private final Counter eventByteReadRate;
+    private final SamzaHistogram readLatency;
+    private final Counter errors;
+    private final Serde<byte[]> serde;
+    SystemStreamPartition ssp;
+
+    PartitionReceiverHandlerImpl(SystemStreamPartition ssp, Counter eventReadRate, Counter eventByteReadRate,
+                                 SamzaHistogram readLatency, Counter readErrors, Serde<byte[]> serde) {
+      super(MAX_EVENT_COUNT_PER_PARTITION_POLL);
+      this.ssp = ssp;
+      this.eventReadRate = eventReadRate;
+      this.eventByteReadRate = eventByteReadRate;
+      this.readLatency = readLatency;
+      errors = readErrors;
+      this.serde = serde;
+    }
+
+    @Override
+    public void onReceive(Iterable<EventData> events) {
+      if (events != null) {
+
+        events.forEach(event -> {
+            byte[] decryptedBody = event.getBytes();
```

Comment:
Remove references to encryption and de-cryption everywhere? since they seem orthogonal to the record payloadss

Reply:
Removed the Serde configs and added the Stream to Serde mapping to `EventHubSystemFactory`. (Passthrough Serde by default)